pkg/kubelet/kuberuntime/instrumented_services_test.go

```diff
@@ -89,3 +90,112 @@ func TestStatus(t *testing.T) {
 	}
 	assert.Equal(t, expected, actural)
 }
+
+func TestInstrumentedListContainer(t *testing.T) {
+	fakeRuntime, _, _, _ := createTestRuntimeManager()
+	irs := newInstrumentedRuntimeService(fakeRuntime)
+	podName, namespace := "foo", "bar"
+	containerName, image := "sidecar", "logger"
+	configs := []*runtimeapi.ContainerConfig{}
+	sConfigs := []*runtimeapi.PodSandboxConfig{}
+	for i := 0; i < 3; i++ {
+		s := makeSandboxConfig(fmt.Sprintf("%s%d", podName, i),
+			fmt.Sprintf("%s%d", namespace, i), fmt.Sprintf("%d", i), 0)
+		labels := map[string]string{"abc.xyz": fmt.Sprintf("label%d", i)}
+		annotations := map[string]string{"foo.bar.baz": fmt.Sprintf("annotation%d", i)}
+		c := makeContainerConfig(s, fmt.Sprintf("%s%d", containerName, i),
+			fmt.Sprintf("%s:v%d", image, i), uint32(i), labels, annotations)
+		sConfigs = append(sConfigs, s)
+		configs = append(configs, c)
+	}
+
+	expected := make(map[string]*runtimeapi.Container, 0)
+	expectedStatus := make(map[string]*runtimeapi.ContainerStatus, 0)
+	actual := make(map[string]*runtimeapi.Container, 0)
+	state := runtimeapi.ContainerState_CONTAINER_RUNNING
+	createdAt := int64(0)
+	for i := range configs {
+		// We don't care about the sandbox id; pass a bogus one.
+		sandboxID := fmt.Sprintf("sandboxid%d", i)
+		id, err := irs.CreateContainer(sandboxID, configs[i], sConfigs[i])
+		assert.NoError(t, err)
+		err = irs.StartContainer(id)
+		assert.NoError(t, err)
+		expected[id] = &runtimeapi.Container{
+			Metadata:     configs[i].Metadata,
+			Id:           id,
+			PodSandboxId: sandboxID,
+			State:        state,
+			CreatedAt:    createdAt,
+			Image:        configs[i].Image,
+			ImageRef:     configs[i].Image.Image,
+			Labels:       configs[i].Labels,
+			Annotations:  configs[i].Annotations,
+		}
+
+		expectedStatus[id] = &runtimeapi.ContainerStatus{
+			Metadata:    configs[i].Metadata,
+			State:       runtimeapi.ContainerState_CONTAINER_EXITED,
+			Id:          id,
+			ExitCode:    0,
+			Image:       configs[i].Image,
+			ImageRef:    configs[i].Image.Image,
+			Labels:      configs[i].Labels,
+			Annotations: configs[i].Annotations,
+		}
+	}
+	containers, err := irs.ListContainers(nil)
+	assert.NoError(t, err)
+	for _, c := range containers {
+		c.CreatedAt = createdAt
+		actual[c.Id] = c
+	}
+	// assert.Len(t, actual, len(expected))
+	assert.Equal(t, actual, expected)
+	for id := range expected {
+		err = irs.StopContainer(id, 0)
+		assert.NoError(t, err)
+		status, err := irs.ContainerStatus(id)
+		assert.NoError(t, err)
+		expectedStatus[id].CreatedAt = status.CreatedAt
```

Comment:
can you add a comment explaining why we need to set these values explicitly?  How are these values set normally?